modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java

```diff
@@ -334,14 +328,124 @@ public void runScript() {
+            alert.setHeaderText("In order to insert missing case statements the script has to be run first.");
+            alert.showAndWait();
+            if(alert.getResult() == ButtonType.OK) {
+                runScript();
+            } else {
+                return;
+            }
+        }
+        ProofScript script = this.proof.getProofScript();
+        Statements newScript = insertCasesForStatement(proof.getProofRoot(), script.getBody());
+        script.setBody(newScript);
+        PrettyPrinter pp = new PrettyPrinter();
+        script.accept(pp);
+        view.replaceText(pp.toString());
+        runScript();
+    }
+
+    /**@
+     * recursivly inserts all missing case statements in the given script
+     *
+     * @param pn the proofnode for which the cases should be inserted
+     * @param stmts the current script that should be extended by the missing case statements
+     * @return a new script containing all necessary case statements
+     */
+    private Statements insertCasesForStatement(ProofNode pn, Statements stmts) {
+        if(stmts.size() == 0) {
+            return stmts;
+        }
+        Statements result = new Statements();
+        for (int i = 0; i < stmts.size() - 1; ++i) {
+            if(stmts.get(i) instanceof CallStatement) {
+                result.add(stmts.get(i));
+            } else {
+                Logger.getGlobal().warning("Only the last statement may be a cases-statement.");
+                return stmts;
+            }
+            if((pn.getChildren() != null && pn.getChildren().size() == 1) || i == stmts.size() - 2) {
+                if(i != stmts.size() - 2) {
+                    pn = pn.getChildren().get(0);
+                }
+            } else {
+                Logger.getGlobal().warning("Script has unexpected number of children at some point.");
+                return stmts;
+            }
+        }
+        Statement st = stmts.get(stmts.size() - 1);
+        if(pn.getChildren().size() == 1 && st instanceof CallStatement) {
+            result.add(st);
+        } else if (pn.getChildren().size() > 1 && st instanceof CasesStatement) {
+                result.add(createCasesForNode(pn, ((CasesStatement) st).getCases()));
+        } else if (pn.getChildren().size() > 1 && !(st instanceof CasesStatement)) {
+            result.add(st);
+            result.add(createCasesForNode(pn));
+        }
+
+        return result;
+    }
+
+    /**@
+     * Creates all cases for the given proofnode except the ones given in cases
+     *
+     * @param pn the proofnode for which the cases should be created
+     * @param cases the cases that already exist
+     * @return a case statement containing all necesarry cases
+     */
+    private Statement createCasesForNode(ProofNode pn, List<CaseStatement> cases) {
+        CasesStatement res = new CasesStatement();
+        for(ProofNode p : pn.getChildren()) {
+            boolean found = false;
+            for(CaseStatement caze : cases) {
```

Comment:
Name? Maye case?

Reply:
thats a jaava keyword and thus cant be used as a variable name :)